application: nrf5340_audio: Refine DFU document (#8937)
nrfconnect/sdk-nrf: applications/nrf5340_audio/README.rst

     python buildprog.py -c app -b debug -d both

> * Example 2: The following command builds both cores, using the DFU internal flash memory and the minimal size of the network core bootloader:

Reviewer: Would "using internal flash memory for storing DFU image" be better?

> * Example 2: The following command builds both cores, using the DFU internal flash memory and the minimal size of the network core bootloader:

Author: Revoke this change.

>    .. code-block:: console

Author: Revoke this change.

>    .. code-block:: console

Reviewer 2: It does not look like this was related to your change. Can you try to remove one space so that the command line examples get tabbed the same? :)

>      python buildprog.py -c both -b debug -d both -m internal -M

Author: ``-c app`` also can create signed ble_ctrl.
I will revoke this change.

     python buildprog.py -c app -b debug -d both

> * Example 2: The following command builds the both core, and with the DFU internal flash memory layout enabled and using the minimal size of the network core bootloader:

Reviewer 3: ```suggestion
* Example 2: The following command builds both cores, using the DFU internal flash memory and the minimal size of the network core bootloader:
```

Reviewer 3: I thought we had to use external flash if both cores are updated?

Author: We should build both core if DFU is enabled either internal or external layout.
Both cores can be updated one by one if internal flash.
Some of our customers might want to experience this feature without flash.

Reviewer 3: Ok

      python buildprog.py -c both -b debug -d both -p

   .. note::

>       If you have DFU parameter enabled when building binary, the same parameter should be also assigned when programming.

Author: Revised.

         python buildprog.py -c both -b debug -d both -p --recover-on-fail

   If you want to program firmware that has DFU enabled, you must include the DFU parameters in the command.

>    The command for programming with DFU enable can look as follows:

Reviewer 2: enable = enabled

      python buildprog.py -c both -b debug -d headset --pristine -m external

#. Transfer the generated file to your Android or iOS device.
      a. For multi-image DFU, the file name follows the pattern ``dev_<Role>_build_<Application version>_dfu_application<Revision>.zip``,
      For example :file:`dev_headset_build_debug_dfu_application_0.0.0+0.zip`.

>       b. For single-image DFU, the file for app core follows the pattern ``dev_<Role>_build_<Application version>_app_update_<Revision>.bin``

Author: Revised.

>       The file for net core follows the pattern ``dev_<Role>_build_<Application version>_net_core_app_update_<FW_VERSION>.bin``.

Reviewer 2: ```suggestion
      The file for the network core follows the pattern ``dev_<Role>_build_<Application version>_net_core_app_update_<FW_VERSION>.bin``.
```

>       For example :file:`dev_headset_build_debug_app_update_0.0.0+0.bin`, :file:`dev_headset_build_debug_net_core_app_update_1.bin`.

Reviewer 2: The example name for the application core should be moved to step a.

Author: Multi-image DFU can use these files, zip, app core binary, net core.
Since we want multi-image to be DFU at once. We should choose zip file.
The app core/net core binary should go to single-image DFU description.

Reviewer 4: I don't understand. Does this mean single-image uses 2 files and multi-image uses 1 file? That is totally counterintuitive :/

Reviewer 2: 2 .bin files are being generated for single core DFU (net.bin and app.bin), but only one file .bin file is used at a time

>       The ``<Role>`` is ``headset`` or ``gateway``, ``<Application version>`` is ``debug`` or ``release``, ``<Revision>`` is :kconfig:option:`CONFIG_MCUBOOT_IMAGE_VERSION`,

Reviewer 2: Maybe put this in a note?

``debug`` = ``build_debug``
``release`` = ``build_release``

Add "and" in stead of comma before ``<FW_VERSION>``

      ``<FW_VERSION>`` is :kconfig:option:`CONFIG_FW_INFO_FIRMWARE_VERSION`.

   For transfer, you can use cloud services like Google Drive for Android or iCloud for iOS.
#. Open `nRF Connect Device Manager`_ and look for ``NRF5340_AUDIO_HL_DFU`` in the scanned devices window.
   The headset is left by default.
#. Tap on :guilabel:`NRF5340_AUDIO_HL_DFU` and then on the downward arrow icon at the bottom of the screen.
#. In the :guilabel:`Firmware Upgrade` section, tap :guilabel:`SELECT FILE`.
#. Select the file you transferred to the device.

> #. Tap :guilabel:`START` and check :guilabel:`Confirm only` then :guilabel:`START` again in the notification to start the DFU process.

Reviewer 2: ```suggestion
#. Tap :guilabel:`START` and check :guilabel:`Confirm only` in the notification. Then tap :guilabel:`START` again to start the DFU process.
```

Author: Revised.
